### DISCUSSION TITLE
ESP8266 I2S input stream

### DISCUSSION THREAD

**Discussion by steenl** [pschatzmann/arduino-audio-tools#238] [Q&A] [answered]
I am trying to stream an INMP441 I2C stream out to a web server channel on ESP82166 ESP-12F.  (The self-generated stream such as example streams-generator-webserver-wav works great)

After connecting 3.3V, gnd, I2SI_data=GPIO12, I2SI_BCK=GPIO13, I2SI_WS=GPIO14, L/W=gnd, I see no debug activity after the SSID assignation.  

I am trusting the library sets the ESP8266 I2S GPIO and function settings, but is there a way to verify/check?  (Apologies if I missed in the documentation).

**Comment by steenl** [comment]
A combination of setting below and pressing setting/starting the web server channel got it working 

My intent is to record periodic 30second beehive samples for hive health analysis (queen conflicts, swarming, etc)

Thanks!

 pinMode(12, INPUT);
  pinMode(13, INPUT);
  pinMode(14, INPUT);  
  pinMode(12, FUNCTION_1);
  pinMode(13, FUNCTION_1);
  pinMode(14, FUNCTION_1);

**Comment by pschatzmann** [comment, marked as answer]
Strange: My begin() method for the ESP8288 is calling i2s_rxtx_begin() which is defining the pinMode...
Did you use

I2SStream i2s;
i2s.begin(RX_MODE);

**Reply by steenl** [reply]
Thanks Phil, taking my changes out it still works, so pilot error on my part.
